### PR TITLE
feat(debug): add query json to debug ch queries

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

No problems, only solutions.

## Changes

Adds a button to open the query node directly under `/debug` from the "Debug CH queries" pane, or to copy it:

image

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Opened the modal with different queries.

### CODE REVIEW

File: frontend/src/lib/components/CommandPalette/DebugCHQueries.tsx [hunk 1]
```diff
@@ -120,9 +123,25 @@ function DebugCHQueries(): JSX.Element {
...
-                            <span className="font-mono whitespace-pre">
-                                {dayjs.tz(item.timestamp, 'UTC').tz().format().replace('T', '\n')}
-                            </span>
+                            <div className="space-y-2">
+                                <span className="font-mono whitespace-pre">
+                                    {dayjs.tz(item.timestamp, 'UTC').tz().format().replace('T', '\n')}
+                                </span>
+                                {item.queryJson ? (
+                                    <div className="flex">
+                                        <LemonButton type="primary" to={urls.debugQuery(item.queryJson)}>
```
Comment (Twixes) [inline frontend/src/lib/components/CommandPalette/DebugCHQueries.tsx:132]: Let's make this `targetBlank`, more intuitive than leaving the insight altogether (and safer)

File: frontend/src/lib/components/CommandPalette/DebugCHQueries.tsx [hunk 2]
```diff
@@ -120,9 +123,25 @@ function DebugCHQueries(): JSX.Element {
...
+                                            {JSON.parse(item.queryJson).kind || 'Debug'}
+                                        </LemonButton>
+                                        <CopyToClipboardInline
+                                            iconStyle={{ color: 'var(--lemon-button-icon-opacity)' }}
+                                            className="font-normal text-sm"
+                                            explicitValue={item.queryJson}
+                                            description="Query JSON"
```
Comment (Twixes) [inline frontend/src/lib/components/CommandPalette/DebugCHQueries.tsx:139]: Description should be word-cased (i.e. starting with lowercase)

File: frontend/src/queries/Query/Query.tsx [hunk 3]
```diff
@@ -106,6 +103,8 @@ export function Query(props: QueryProps): JSX.Element | null {
         component = <TimeToSeeData query={query} cachedResults={props.cachedResults} />
     } else if (isWebOverviewQuery(query)) {
         component = <WebOverview query={query} cachedResults={props.cachedResults} context={queryContext} />
+    } else {
```
Comment (Twixes) [inline frontend/src/queries/Query/Query.tsx:106]: Why not keep the `if (isDataNode(query))`? Seemed safer

**Review by webjunkie — APPROVED** [review]

File: frontend/src/queries/Query/Query.tsx (hunk 3 above)
Comment (mariusandra) [reply frontend/src/queries/Query/Query.tsx:106]: Essentially everything else that doesn't have a custom visualization should be a data node. We're just not very good at keeping the `isDataNode` function up to date.

**Comment by Twixes** [comment]
Since I was already looking at this code, I fixed a few things that were slightly annoying me when doing support:
1. Button weirdly overlapping the code due to some broken margins
2. No indication how much of the snippet is collapsed

Also, the timestamp area was really cluttered with this PR's addition, I think the query should belong to the "Query" column.
This should make debugging a bit faster/easier to grok:

### Before

Screenshot 2024-04-18 at 13 17 30

### After
Screenshot 2024-04-18 at 13 09 50

**Pull request merged**